soc/nxp/imx/imx8m/Kconfig.soc

```
@@ -0,0 +1,80 @@
+# Copyright 2020-2022,2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_IMX8M
+	bool
+	select SOC_FAMILY_IMX
+
+config SOC_MIMX8MM
+	bool
+	select SOC_SERIES_IMX8M
+
+config SOC_MIMX8MM_A53
+	bool
+	select SOC_MIMX8MM
+	help
+		NXP i.MX8MM A53
+
+config SOC_MIMX8MP
+	bool
+	select SOC_SERIES_IMX8M
+
+config SOC_MIMX8MP_A53
+	bool
+	select SOC_MIMX8MP
+	help
+		NXP i.MX8MP A53
+
+config SOC_MIMX8MN
+	bool
+	select SOC_SERIES_IMX8M
+
+config SOC_MIMX8MN_A53
+	bool
+	select SOC_MIMX8MN
+	help
+		NXP i.MX8MN A53
+
+config SOC_PART_NUMBER_MIMX8ML8DVNLZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MM6DVTLZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MM6CVTKZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MN6DVTJZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MN6DUCJZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MN6CVTIZ
+	bool
+
+config SOC_PART_NUMBER_MIMX8MN6CUCIZ
+	bool
+
+config SOC_PART_NUMBER
+	default "MIMX8ML8DVNLZ" if  SOC_PART_NUMBER_MIMX8ML8DVNLZ
+	default "MIMX8MM6DVTLZ" if  SOC_PART_NUMBER_MIMX8MM6DVTLZ
+	default "MIMX8MM6CVTKZ" if  SOC_PART_NUMBER_MIMX8MM6CVTKZ
+	default "MIMX8MN6DVTJZ" if  SOC_PART_NUMBER_MIMX8MN6DVTJZ
+	default "MIMX8MN6DUCJZ" if  SOC_PART_NUMBER_MIMX8MN6DUCJZ
+	default "MIMX8MN6CVTIZ" if  SOC_PART_NUMBER_MIMX8MN6CVTIZ
+	default "MIMX8MN6CUCIZ" if  SOC_PART_NUMBER_MIMX8MN6CUCIZ
+
+config SOC
+	default "mimx8mm6" if SOC_MIMX8MM_A53
```

```suggestion
	default "mimx8mm6" if SOC_MIMX8MM
```

Also, why does the `CONFIG_SOC` value not match the SOC Kconfig name? IE why is the SOC Kconfig name not `SOC_MIMX8MM6`?

---

Aligned CONFIG_SOC and SoC Kconfig name, thanks.